packages/java-parser/test/samples-spec.js

```diff
@@ -32,3 +33,39 @@ function createSampleSpecs(sampleName) {
     });
   });
 }
+
+function createFailingSampleSpec(sampleName, maxFailAllowed, timeout) {
+  context(sampleName + " samples", () => {
+    const samplesDir = path.resolve(__dirname, "../samples/" + sampleName);
+    const sampleFiles = klawSync(samplesDir, { nodir: true });
+    const javaSampleFiles = sampleFiles.filter(fileDesc =>
+      fileDesc.path.endsWith(".java")
+    );
+
+    if (_.isEmpty(sampleFiles)) {
+      throw `Missing sample-dir: <${samplesDir}> did you forget to clone the samples?`;
+    }
+
+    const errors = [];
+    it(`Can Parse ${sampleName} repository`, function() {
+      if (timeout !== undefined) {
+        this.timeout(timeout);
+      }
+
+      _.forEach(javaSampleFiles, fileDesc => {
+        const relativePath = path.relative(__dirname, fileDesc.path);
+        const sampleText = fs.readFileSync(fileDesc.path, "utf8");
+
+        try {
+          javaParser.parse(sampleText);
+        } catch (error) {
+          errors.push(relativePath);
+        }
+      });
+
+      expect(errors.length, `Can not Parse ${errors}`).to.be.at.most(
```

Comment:
lets make it **exactly** the number of expected errors, so each time we fix a bug
we would have to adjust the number of failures downwards.